### ISSUE TITLE
JWT vulnerability with old version of bshaffer/oauth2-server-php as a dependency

### ISSUE THREAD

**Issue by alexkb** [Luracast/Restler#519]
As per bshaffer/oauth2-server-php#564 - there was an update [back in April 2015](https://github.com/bshaffer/oauth2-server-php/commit/535d658dfc54a8aa990061223993cc68fc2a8695#diff-09f8deffcefa5424956402f806d15f15) to fix a vulnerability with JWT. It seems Restler (by default) still uses an [old version of oauth2-server-php](https://github.com/Luracast/Restler/blob/master/composer.json#L48), meaning the security vulnerability still exists in Restler.

Is there any plans to update to a newer version of oauth2-server?

**Comment by thedotedge** [comment]
:+1:

**Comment by alexkb** [comment]
As a temporary workaround we've simply switched to using the latest [F21/jwt](https://github.com/F21/jwt) class. All we did was:

    $ composer require F21/jwt
    $ composer dump-autoload

From your AccessControl class, simply:

    -use OAuth2\Encryption\Jwt;
    +use JWT;

Then change your calls to `JWT()` as per the updated method parameters.

Also note that stock F21/jwt returns a class in its `decode()` method. This must of been a change @Arul- did in Restler's copy of the JWT class as I can't see it ever being made in the [F21/JWT history](https://github.com/F21/jwt/commits/master/JWT/JWT.php) (simply changed back by adjusting the `json_decode()` calls).
